virttest/vt_console.py

```diff
@@ -60,6 +69,15 @@ def __login(self, linesep, status_test_command,
                 prompt, username, password, timeout):
         self._console.set_linesep(linesep)
         self._console.set_status_test_command(status_test_command)
+
+        end_time = time.time() + timeout
+        while time.time() < end_time:
+            if self._console.is_responsive():
+                self.is_responsive = True
```

Comment:
Hi @chloerh is there some reason to make this as an attribute of the `ConsoleManager` class?

Reply:
Hi @luckyh . My thought was there was going to be a flag to mark the responsive status of console, and it occurred to me that this flag might be used by other functions of console. Then I decided to put it to the attribute list.

Thanks for reviewing. Actually this was kind of a workaround. Have you guys met this 'Reseting System' issue(Press any key to stop resetting system)? We would very much appreciate your suggestions or any other solutions. It has been bothering us for at least 2 weeks. Thanks again.

Reply:
> Hi @luckyh . My thought was there was going to be a flag to mark the responsive status of console, and it occurred to me that this flag might be used by other functions of console. Then I decided to put it to the attribute list.

I see. However, I have the concerns below about that

* Because the `ConsoleManager` is the class that you're going to have the attribute associated with, I'm wondering if it would make sense about the meaning of "console-manager is responsive" (since IMO, console-managers should always be responsive once created). Therefore, I'd like to request a better definition to that attribute.
* Due to consoles may change the state of responsiveness from time to time, it's not suggested for setting a static value as the cache.
 
> Thanks for reviewing. Actually this was kind of a workaround. Have you guys met this 'Reseting System' issue(Press any key to stop resetting system)? We would very much appreciate your suggestions or any other solutions. It has been bothering us for at least 2 weeks. Thanks again.

Uh, I haven't heard about this issue from the internal qemu testing users so far. Btw, previously we have solved a very similar issue (needing a confirmation by user to continue to boot from cdrom) by adding a step to call `vm.send_key` at boot stage, but I'm not sure if that would help for the case you are handling.

Reply:
> > Hi @luckyh . My thought was there was going to be a flag to mark the responsive status of console, and it occurred to me that this flag might be used by other functions of console. Then I decided to put it to the attribute list.
> 
> I see. However, I have the concerns below about that
> 
> * Because the `ConsoleManager` is the class that you're going to have the attribute associated with, I'm wondering if it would make sense about the meaning of "console-manager is responsive" (since IMO, console-managers should always be responsive once created). Therefore, I'd like to request a better definition to that attribute.
> * Due to consoles may change the state of responsiveness from time to time, it's not suggested for setting a static value as the cache.
> 

You're right. Thanks for the detailed comment. I'll change it to a local variable.

> > Thanks for reviewing. Actually this was kind of a workaround. Have you guys met this 'Reseting System' issue(Press any key to stop resetting system)? We would very much appreciate your suggestions or any other solutions. It has been bothering us for at least 2 weeks. Thanks again.
> 
> Uh, I haven't heard about this issue from the internal qemu testing users so far. Btw, previously we have solved a very similar issue (needing a confirmation by user to continue to boot from cdrom) by adding a step to call `vm.send_key` at boot stage, but I'm not sure if that would help for the case you are handling.

It does sound similar. I'll look it up.

Thanks again. I'll make the update a little later.